cps-global-lib/src/main/java/org/jenkinsci/plugins/workflow/cps/global/UserDefinedGlobalVariable.java

```diff
@@ -0,0 +1,84 @@
+package org.jenkinsci.plugins.workflow.cps.global;
+
+import groovy.lang.Binding;
+import hudson.markup.EscapedMarkupFormatter;
+import org.apache.commons.io.Charsets;
+import org.apache.commons.io.FileUtils;
+import org.jenkinsci.plugins.workflow.cps.CpsScript;
+import org.jenkinsci.plugins.workflow.cps.CpsThread;
+import org.jenkinsci.plugins.workflow.cps.GlobalVariable;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * Global variable backed by user-supplied script.
+ *
+ * @author Kohsuke Kawaguchi
+ * @see UserDefinedGlobalVariableList
+ */
+// not @Extension because these are instantiated programmatically
+public class UserDefinedGlobalVariable extends GlobalVariable {
+    private final WorkflowLibRepository repo;
+    private final String name;
+
+    /*package*/ UserDefinedGlobalVariable(WorkflowLibRepository repo, String name) {
+        this.repo = repo;
+        this.name = name;
+    }
+
+    @Nonnull
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Nonnull
+    @Override
+    public Object getValue(@Nonnull CpsScript script) throws Exception {
+        Binding binding = script.getBinding();
+        Object instance;
+        if (binding.hasVariable(getName())) {
+            instance = binding.getVariable(getName());
+        } else {
+            CpsThread c = CpsThread.current();
+            if (c==null)
+                throw new IllegalStateException("Expected to be called from CpsThread");
+
+            instance = c.getExecution().getShell().getClassLoader().parseClass(source(".groovy")).newInstance();
+            binding.setVariable(getName(), instance);
+        }
+        return instance;
+    }
+
+    /**
+     * Loads help from user-defined file, if available.
+     */
+    public String getHelpHtml() throws IOException {
+        File help = source(".txt");
+        if (!help.exists())     return null;
+
+        return new EscapedMarkupFormatter().translate(FileUtils.readFileToString(help, Charsets.UTF_8));
```

Comment:
This is simply a roundabout way of saying `Util.escape`. But do we not want to use `Jenkins.getInstance().getMarkupFormatter().translate(…)` instead?

(The `*.txt` extension would not always be appropriate in that case, of course, but close enough.)